src/client/MapleCharacter.java

```diff
@@ -1806,14 +1795,28 @@ public void saveToDB(boolean dc, boolean fromcs) {
                     }
                 }
             }
-            changed_wishlist = false;
+			
+            if (changed_favorites) {
+                ps = con.prepareStatement("DELETE FROM favorites WHERE accountid = ?");
+                ps.setInt(1, getAccountID());
+                ps.executeUpdate();
+                ps.close();
+                for (int i : favorites) {
+                    ps = con.prepareStatement("INSERT INTO favorites(accountid, sn) VALUES(?, ?) ");
```

Comment:
I was initially confused with this, but it does make sense to have the prepareStatement and close outside of the loop. I stuck with the same pattern among others in CharacterTransfer for the consistency for now, though.